mbed-client-libservice/ip6string.h

```diff
@@ -18,14 +18,25 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
+
+#include "ns_types.h"
 /**
  * Print binary IPv6 address to a string.
  * String must contain enough room for full address, 40 bytes exact.
  * IPv4 tunneling addresses are not covered.
  * \param ip6addr IPv6 address.
  * \p buffer to write string to.
  */
-void ip6tos(const void *ip6addr, char *p);
+uint8_t ip6tos(const void *ip6addr, char *p);
```

Comment:
If you don't want to return size_t (and there are reasons not to on small platforms), use uint_fast8_t instead. Saves bigger platforms messing around trying to makes sure it's narrowed - they can just return 32-bit.